### PR TITLE
Schematic Drill Indicator

### PR DESCRIPTION
**Pull request by zethnest**
This implements https://github.com/Anuken/Mindustry-Suggestions/issues/165

screenshot200517011457749952295
screenshot200517011503916301425

Two pictures above shows the mined ore + the mining speed

screenshot200517004303805256764

Picture above only shows the mined ore.

### CODE REVIEW

**Comment by zethnest** [comment]
Current implementation as of e454199d9 looks like this

**Comment by Anuken** [comment]
The item colors are useful, but I don't think numbers should be displayed. They create a lot of visual noise, especially when drills are placed in clusters.

File: core/src/mindustry/world/blocks/production/Drill.java [hunk 1]
```diff
@@ -73,57 +71,13 @@ public Drill(String name){
...
-        font.setUseIntegerPositions(ints);
-        font.setColor(Color.white);
-        font.getData().setScale(1f);
-        Draw.reset();
-        Pools.free(layout);
-        /* end of code section */
-
-        /* This commented code colors their ore region */
         Draw.color(returnItem.color);
         Draw.rect("drill-top", req.drawx(), req.drawy(), 1f);
```
Comment (Anuken) [inline core/src/mindustry/world/blocks/production/Drill.java:79]:
```suggestion
        Draw.rect(topRegion, req.drawx(), req.drawy(), 1f);
```

File: core/src/mindustry/world/blocks/production/Drill.java [hunk 2]
```diff
@@ -66,6 +67,19 @@ public Drill(String name){
...
+    @Override
+    public void drawRequestConfigTop(BuildRequest req, Eachable<BuildRequest> list){
+        Tile tile = req.tile();
+        if(tile == null) return;
+
+        countOre(req.tile());
+        if(returnItem == null) return;
+
+        Draw.color(returnItem.color);
+        Draw.rect(topRegion, req.drawx(), req.drawy(), 1f);
```
Comment (Anuken) [inline core/src/mindustry/world/blocks/production/Drill.java:79]: Aren't you drawing it with 1 degree of rotation here?

File: core/src/mindustry/world/blocks/production/Drill.java [hunk 3]
```diff
@@ -66,6 +67,19 @@ public Drill(String name){
         idleSoundVolume = 0.003f;
     }
 
+    @Override
+    public void drawRequestConfigTop(BuildRequest req, Eachable<BuildRequest> list){
+        Tile tile = req.tile();
+        if(tile == null) return;
```
Comment (Anuken) [inline core/src/mindustry/world/blocks/production/Drill.java:73]: Technical fix: You should also skip drawing if `!req.worldContext` (so schematics don't grab world state).

File: core/src/mindustry/world/blocks/production/Drill.java (hunk 1 above)
Comment (zethnest) [reply core/src/mindustry/world/blocks/production/Drill.java]: I didn't check before. but `topRegion` makes them look like they have a mole.
screenshot200517121852617205364

File: core/src/mindustry/world/blocks/production/Drill.java (hunk 2 above)
Comment (zethnest) [reply core/src/mindustry/world/blocks/production/Drill.java:79]: I didn't notice before as I copied it from `draw()`. Removing `1f` doesn't seem to change anything so I'll remove that and from `draw()`

**Pull request merged**